src/RamExecutor.cpp

```diff
@@ -351,13 +351,34 @@ bool eval(const RamCondition& cond, RamEnvironment& env, const EvalContext& ctxt
                     }
                     return result;
                 }
+                case BinaryConstraintOp::NOT_MATCH: {
+                    RamDomain l = eval(relOp.getLHS(), env, ctxt);
+                    RamDomain r = eval(relOp.getRHS(), env, ctxt);
+                    const std::string& pattern = env.getSymbolTable().resolve(l);
```

Comment:
This seems like a bad idea - you're making a const string reference to the result of method that returns a const char*. What is actually a reference to?

It does seem to work, though, and is a copy paste of the existing code, so maybe I'm missing something.

Reply:
So, this is fine. A temporary string is constructed and the const ref can use that. It still seems like a bad idea to me, but it's also valid C++ and the approach we already use.